From 53125afb98a3edfcf59cb543d3fd5a1fbce74783 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Wed, 5 Feb 2025 13:25:28 +0100
Subject: [PATCH 18/18] arch: arm: dts: cn9130-sr-som / cn9130-sr-cex7: fix
 net, rtc and pcie

Patch device-tree with u-boot specific driver quirks to fix various
issues:

1. pcie card detection on soft reboot due to reset-gpio toggle

2. dhcp/ping commands on rgmii interfaces timing out due to wrong phy
   configuration

3. date command due to defautl status of cp0_rtc was disabled

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm/dts/cn9130-cf-base-u-boot.dtsi     | 19 ++++++++++++++
 arch/arm/dts/cn9130-cf-pro-u-boot.dtsi      | 28 +++++++++++++++++++++
 arch/arm/dts/cn9131-cf-solidwan-u-boot.dtsi |  5 ++++
 arch/arm/dts/cn9132-clearfog-u-boot.dtsi    | 16 ++++++++++++
 4 files changed, 68 insertions(+)

diff --git a/arch/arm/dts/cn9130-cf-base-u-boot.dtsi b/arch/arm/dts/cn9130-cf-base-u-boot.dtsi
index 0080896b7f9..db310dbe852 100644
--- a/arch/arm/dts/cn9130-cf-base-u-boot.dtsi
+++ b/arch/arm/dts/cn9130-cf-base-u-boot.dtsi
@@ -4,3 +4,22 @@
 		tlv1 = &carrier_eeprom;
 	};
 };
+
+&cp0_eth2 {
+	/* u-boot Marvell 88E151x driver does not understand "rgmii-id" */
+	phy-mode = "rgmii";
+};
+
+&cp0_pcie2 {
+	/*
+	 * Unlike the Linux dw-pcie driver, u-boot pcie_dw_mvebu
+	 * uses non-standard "marvell,reset-gpio" property
+	 * and does not invert polarity internally.
+	 */
+	marvell,reset-gpio = <&expander0 1 GPIO_ACTIVE_LOW>;
+};
+
+&cp0_rtc {
+	/* unlike upstream dts this rtc is disabled by default */
+	status = "okay";
+};
diff --git a/arch/arm/dts/cn9130-cf-pro-u-boot.dtsi b/arch/arm/dts/cn9130-cf-pro-u-boot.dtsi
index 0080896b7f9..2a39763ed19 100644
--- a/arch/arm/dts/cn9130-cf-pro-u-boot.dtsi
+++ b/arch/arm/dts/cn9130-cf-pro-u-boot.dtsi
@@ -4,3 +4,31 @@
 		tlv1 = &carrier_eeprom;
 	};
 };
+
+&cp0_eth2 {
+	/* u-boot Marvell 88E151x driver does not understand "rgmii-id" */
+	phy-mode = "rgmii";
+};
+
+&cp0_pcie1 {
+	/*
+	 * Unlike the Linux dw-pcie driver, u-boot pcie_dw_mvebu
+	 * uses non-standard "marvell,reset-gpio" property
+	 * and does not invert polarity internally.
+	 */
+	marvell,reset-gpio = <&expander0 2 GPIO_ACTIVE_LOW>;
+};
+
+&cp0_pcie2 {
+	/*
+	 * Unlike the Linux dw-pcie driver, u-boot pcie_dw_mvebu
+	 * uses non-standard "marvell,reset-gpio" property
+	 * and does not invert polarity internally.
+	 */
+	marvell,reset-gpio = <&expander0 1 GPIO_ACTIVE_LOW>;
+};
+
+&cp0_rtc {
+	/* unlike upstream dts this rtc is disabled by default */
+	status = "okay";
+};
diff --git a/arch/arm/dts/cn9131-cf-solidwan-u-boot.dtsi b/arch/arm/dts/cn9131-cf-solidwan-u-boot.dtsi
index 0080896b7f9..2fb393921dd 100644
--- a/arch/arm/dts/cn9131-cf-solidwan-u-boot.dtsi
+++ b/arch/arm/dts/cn9131-cf-solidwan-u-boot.dtsi
@@ -4,3 +4,8 @@
 		tlv1 = &carrier_eeprom;
 	};
 };
+
+&cp0_rtc {
+	/* unlike upstream dts this rtc is disabled by default */
+	status = "okay";
+};
diff --git a/arch/arm/dts/cn9132-clearfog-u-boot.dtsi b/arch/arm/dts/cn9132-clearfog-u-boot.dtsi
index 8675c493d4d..47d0f28140b 100644
--- a/arch/arm/dts/cn9132-clearfog-u-boot.dtsi
+++ b/arch/arm/dts/cn9132-clearfog-u-boot.dtsi
@@ -1,3 +1,14 @@
+/ {
+	aliases {
+		tlv0 = &com_eeprom;
+	};
+};
+
+&cp0_eth1 {
+	/* u-boot Marvell 88E151x driver does not understand "rgmii-id" */
+	phy-mode = "rgmii";
+};
+
 &cp0_gpio1 {
 	pinctrl-names= "default";
 	pinctrl-0 = <&cp0_fan_pwm_pins>;
@@ -10,3 +21,8 @@
 		line-name = "fan-pwm";
 	};
 };
+
+&cp0_rtc {
+	/* unlike upstream dts this rtc is disabled by default */
+	status = "okay";
+};
-- 
2.43.0

